Initial quantile operator implementation

heitorschueroff opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42755 Initial quantile operator implementation**

Attempting to land quantile again after being landed here https://github.com/pytorch/pytorch/pull/39417 and reverted here https://github.com/pytorch/pytorch/pull/41616.

Differential Revision: [D23030338](https://our.internmc.facebook.com/intern/diff/D23030338)

heitorschueroff committed:
Initial quantile operator implementation

Author:    Heitor Schueroff <heitorschueroff@fb.com>
Date:      Mon Jun 8 09:18:02 2020 -0700

[ghstack-poisoned]

mruberry commented on torch/_torch_docs.py:
```diff
@@ -4014,6 +4014,70 @@ def merge_dicts(*dicts):
+    >>> torch.quantile(a, q)
+    tensor([-0.5446,  0.0700,  0.9214])
+
+.. function:: quantile(input, q, dim=None, keepdim=False, out=None) -> Tensor
```
Nit: keepdim=False, *, out=None

mruberry reviewed (approved):
Stamped! Looking forward to having this back in the build.

heitorschueroff committed:
Update on "Initial quantile operator implementation"


Attempting to land quantile again after being landed here https://github.com/pytorch/pytorch/pull/39417 and reverted here https://github.com/pytorch/pytorch/pull/41616.

[ghstack-poisoned]
